src/openxr/OpenXRApi.cpp

```diff
@@ -2898,27 +2989,54 @@ XrTime OpenXRApi::get_next_frame_time() const {
 
 XrResult OpenXRApi::acquire_image(int eye) {
 	XrResult result;
-	XrSwapchainImageAcquireInfo swapchainImageAcquireInfo = {
-		.type = XR_TYPE_SWAPCHAIN_IMAGE_ACQUIRE_INFO, .next = nullptr
+
+	if (acquired_image) {
+		Godot::print_error(
+				String("OpenXR called acquire image before release"),
+				__FUNCTION__,
+				__FILE__,
+				__LINE__);
+		return XR_ERROR_CALL_ORDER_INVALID;
+	}
+
+	XrSwapchainImageAcquireInfo swapchain_image_acquire_info = {
+		.type = XR_TYPE_SWAPCHAIN_IMAGE_ACQUIRE_INFO,
+		.next = nullptr
 	};
-	result = xrAcquireSwapchainImage(swapchains[eye], &swapchainImageAcquireInfo, &buffer_index[eye]);
-	if (!xr_result(result, "failed to acquire swapchain image!")) {
+	result = xrAcquireSwapchainImage(color_swapchains[eye], &swapchain_image_acquire_info, &buffer_index[eye]);
+	if (!xr_result(result, "failed to acquire swapchain color image!")) {
 		return result;
 	}
 
-	XrSwapchainImageWaitInfo swapchainImageWaitInfo = {
+	XrSwapchainImageWaitInfo swapchain_image_wait_info = {
 		.type = XR_TYPE_SWAPCHAIN_IMAGE_WAIT_INFO,
 		.next = nullptr,
 		.timeout = 17000000, /* timeout in nanoseconds */
 	};
-	result = xrWaitSwapchainImage(swapchains[eye], &swapchainImageWaitInfo);
-	if (!xr_result(result, "failed to wait for swapchain image!")) {
+	result = xrWaitSwapchainImage(color_swapchains[eye], &swapchain_image_wait_info);
+	if (!xr_result(result, "failed to wait for swapchain color image!")) {
 		return result;
 	}
+
+	acquired_image = true;
```

Comment:
Should this flag be per eye?

Reply:
No, because we don't start rendering the second eye before the first is finished and by this time the we have released it.